internal/services/mssql/mssql_virtual_machine_resource_test.go

```diff
@@ -675,6 +697,10 @@ resource "azurerm_mssql_virtual_machine" "test" {
       log_file_size_mb  = 512
     }
   }
+
+  depends_on = [
```

Comment:
Sql VM should be depending on the data disk attachment so as to configure the storage configuration.